tools/retest/main_integration_test.go

```diff
@@ -200,3 +201,26 @@ func TestIntegration(t *testing.T) {
 	assert.NoError(t, err)
 
 }
+
+//go:embed testdata/statuses.json
+var statusesResponse []byte
+
+func TestGetStatuses(t *testing.T) {
+	var server *httptest.Server
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		t.Logf("%s %s", r.Method, r.RequestURI)
+		_, err := w.Write(statusesResponse)
+		assert.NoError(t, err)
+	}
+	server = httptest.NewServer(http.HandlerFunc(handler))
+	t.Cleanup(server.Close)
+
+	client := github.NewClient(server.Client())
+	parse, err := url.Parse(server.URL + "/")
+	assert.NoError(t, err)
+	client.BaseURL = parse
+
+	statuses, err := statusesForPR(context.Background(), client, server.URL)
+	assert.NoError(t, err)
+	assert.Equal(t, map[string]string{"gke-upgrade-tests": "success"}, statuses)
```

Comment:
The description tells about counting only the latest status.
If I read correctly the response objects (maybe trimming the payload to the fields relevant for the test could help), there is:
```
[
  {
    "state": "success",
    "target_url": "https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/10785/pull-ci-stackrox-stackrox-master-gke-upgrade-tests/1780608599945383936",
    "context": "ci/prow/gke-upgrade-tests",
    "created_at": "2024-04-17T16:40:38Z",
    "updated_at": "2024-04-17T16:40:38Z"
  },
  {
    "state": "pending",
    "target_url": "https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/10785/pull-ci-stackrox-stackrox-master-gke-upgrade-tests/1780608599945383936",
    "context": "ci/prow/gke-upgrade-tests",
    "created_at": "2024-04-17T14:46:15Z",
    "updated_at": "2024-04-17T14:46:15Z",
  },
]
```
The one with `"pending"` status seems more recent to me.

Reply:
my bad... 14:46 vs 16:40 🤦